src/target/adiv5.h

```diff
@@ -23,143 +23,165 @@
-#define ADIV5_AP_CSW_HPROT1		(1u << 25)
-#define ADIV5_AP_CSW_SPIDEN		(1u << 23)
+#define ADIV5_AP_CSW_MASTERTYPE_DEBUG (1u << 29)
+#define ADIV5_AP_CSW_HPROT1           (1u << 25)
+#define ADIV5_AP_CSW_SPIDEN           (1u << 23)
 /* Bits 22:12 - Reserved */
 /* Bits 11:8 - Mode, must be zero */
-#define ADIV5_AP_CSW_TRINPROG		(1u << 7)
-#define ADIV5_AP_CSW_DEVICEEN		(1u << 6)
-#define ADIV5_AP_CSW_ADDRINC_NONE	(0u << 4)
-#define ADIV5_AP_CSW_ADDRINC_SINGLE	(1u << 4)
-#define ADIV5_AP_CSW_ADDRINC_PACKED	(2u << 4)
-#define ADIV5_AP_CSW_ADDRINC_MASK	(3u << 4)
+#define ADIV5_AP_CSW_TRINPROG       (1u << 7)
+#define ADIV5_AP_CSW_DEVICEEN       (1u << 6)
+#define ADIV5_AP_CSW_ADDRINC_NONE   (0u << 4)
+#define ADIV5_AP_CSW_ADDRINC_SINGLE (1u << 4)
+#define ADIV5_AP_CSW_ADDRINC_PACKED (2u << 4)
+#define ADIV5_AP_CSW_ADDRINC_MASK   (3u << 4)
 /* Bit 3 - Reserved */
-#define ADIV5_AP_CSW_SIZE_BYTE		(0u << 0)
-#define ADIV5_AP_CSW_SIZE_HALFWORD	(1u << 0)
-#define ADIV5_AP_CSW_SIZE_WORD		(2u << 0)
-#define ADIV5_AP_CSW_SIZE_MASK		(7u << 0)
+#define ADIV5_AP_CSW_SIZE_BYTE     (0u << 0)
+#define ADIV5_AP_CSW_SIZE_HALFWORD (1u << 0)
+#define ADIV5_AP_CSW_SIZE_WORD     (2u << 0)
+#define ADIV5_AP_CSW_SIZE_MASK     (7u << 0)
 
 /* AP Debug Base Address Register (BASE) */
-#define ADIV5_AP_BASE_BASEADDR		(0xFFFFF000u)
-#define ADIV5_AP_BASE_PRESENT		(1u << 0)
-
+#define ADIV5_AP_BASE_BASEADDR (0xFFFFF000u)
+#define ADIV5_AP_BASE_PRESENT  (1u << 0)
 
 /* ADIv5 Class 0x1 ROM Table Registers */
-#define ADIV5_ROM_MEMTYPE			0xFCC
-#define ADIV5_ROM_MEMTYPE_SYSMEM	(1u << 0)
-#define ADIV5_ROM_ROMENTRY_PRESENT  (1u << 0)
-#define ADIV5_ROM_ROMENTRY_OFFSET	(0xFFFFF000u)
-
+#define ADIV5_ROM_MEMTYPE          0xFCC
+#define ADIV5_ROM_MEMTYPE_SYSMEM   (1u << 0)
+#define ADIV5_ROM_ROMENTRY_PRESENT (1u << 0)
+#define ADIV5_ROM_ROMENTRY_OFFSET  (0xFFFFF000u)
 
 /* Constants to make RnW parameters more clear in code */
-#define ADIV5_LOW_WRITE		0
-#define ADIV5_LOW_READ		1
+#define ADIV5_LOW_WRITE 0
+#define ADIV5_LOW_READ  1
 
 #define SWDP_ACK_OK    0x01
 #define SWDP_ACK_WAIT  0x02
 #define SWDP_ACK_FAULT 0x04
 
+/* JEP-106 code list
+ * JEP-106 is a JEDEC standard assigning manufacturer IDs to different manufacturers
+ * the codes in this list are encoded as 16 bit values,
+ * with the first bit marking a legacy code (ASCII, not JEP106), the following 3 bits being NULL/unused
+ * the following 4 bits the number of continuation codes (see JEP106 continuation scheme),
+ * and the last 8 bits being the code itself (without parity, bit 7 is always 0).
+ *
+ * |15     |11     |7|6           0|
+ * | | | | | | | | |0| | | | | | | |
+ *  |\____/ \______/|\_____________/
+ *  |  V        V   |       V
+ *  | Unused   Cont	|      code
+ *  |          Code |
+ *  \_ Legacy flag  \_ Parity bit (always 0)
+ */
+#define ASCII_CODE_FLAG    (1 << 15) /* flag the code as legacy ASCII */
+
+#define JEP106_ARM          0x43B /* ARM Ltd. */
```

Comment:
This is good and a much needed bit of cleanup, however it would be useful in-context if "DESIGNER" or "MANUFACTURER" was somewhere in the constant name - eg `JEP106_DESIGNER_ARM` to provide a stronger association to these definitely being `ap_designer` codes. This also suggests that `ap_designer` should be renamed to `ap_jep106_designer` or similar

Reply:
hm, the adiv5 spec does name it designer though, ugh

Reply:
I'm going to keep `designer` internally for variables and so, but name the macros `JEP106_MANUFACTURER_xxx`, they don't tie as neatly this was, but they both better align with their respective standards